cms/static/js/views/import.js

```diff
@@ -2,241 +2,306 @@
-            } else {
-                // In progress
-                updateStage(currentStage);
+        var updateFeedbackList = function (currStageMsg) {
+            var $checkmark = $dom.successStage.find('.icon');
+            currStageMsg = currStageMsg || '';
+
+            function completeStage(stage) {
+                $(stage)
+                    .removeClass("is-not-started is-started")
+                    .addClass("is-complete");
             }
 
-            var time = timeout || 1000;
-            $.getJSON(url,
-                function (data) {
-                    setTimeout(function () {
-                        getStatus(url, time, data.ImportStatus);
-                    }, time);
-                }
-            );
+            function resetStage(stage) {
+                $(stage)
+                    .removeClass("is-complete is-started has-error")
+                    .addClass("is-not-started")
+                    .find('p.error').remove().end()
+                    .find('p.copy').show();
+            }
+
+            switch (current.state) {
+                case STATE.READY:
+                    _.map($dom.stages, resetStage);
+
+                    break;
+
+                case STATE.IN_PROGRESS:
+                    var $prev = $dom.stages.slice(0, current.stage);
+                    var $curr = $dom.stages.eq(current.stage);
+
+                    _.map($prev, completeStage);
+                    $curr.removeClass("is-not-started").addClass("is-started");
+
+                    break;
+
+                case STATE.SUCCESS:
+                    var successUnix = CourseImport.storedImport().date;
+                    var date = moment(successUnix).utc().format('MM/DD/YYYY');
+                    var time = moment(successUnix).utc().format('HH:mm');
+
+                    _.map($dom.stages, completeStage);
+
+                    $dom.successStage
+                        .find('.item-progresspoint-success-date')
+                        .html('(' + date + ' at ' + time + ' UTC)');
+
+                    break;
+
+                case STATE.ERROR:
+                    // Make all stages up to, and including, the error stage 'complete'.
+                    var $prev = $dom.stages.slice(0, current.stage + 1);
+                    var $curr = $dom.stages.eq(current.stage);
+                    var $next = $dom.stages.slice(current.stage + 1);
+                    var error = currStageMsg || gettext("There was an error with the upload");
+
+                    _.map($prev, completeStage);
+                    _.map($next, resetStage);
+
+                    if (!$curr.hasClass('has-error')) {
+                        $curr
+                            .removeClass('is-started')
+                            .addClass('has-error')
+                            .find('p.copy')
+                            .hide()
+                            .after("<p class='copy error'>" + error + "</p>");
+                    }
```

Comment:
@AkA84 nit: this block would make a nice `errorStage` function - just to be in-line with `completeStage` and `resetStage`